templates/navbar.njk

```diff
@@ -0,0 +1,135 @@
+        </nav>
+    </div>
+{% endmacro %}
+
+{# 
+A macro that adds a menu navigation bar with predefined items.
+It receives:
+- userDisplayText: a text identifying the logged user, e.g. email address or name
+- chsMonitorGuiUrl: the URL for the 'Companies you follow' link
+- lang: an optional object with hidden texts and aria labels. If not provided,
+    default texts are used instead. It contains the following keys:
+    - mainNavbar: aria label text for the nav element, default 'The main navigation bar'
+    - clickToExpandOrCollaps: text for the visually hidden part of the anchor text,
+        default 'Click to expand or collapse'
+    - signInOrRegister: the mobile version of sign in anchor, default 'Sign in / Register'
+
+The navbar requires the use of styles provided in the following stylesheet
+(you can either add the provided link to the head section or include specific
+styles in your own stylesheet)
+
+    <link href="{{ cdnHost }}/stylesheets/ch.gov.uk.css" rel="stylesheet"/>
+
+It also requires the following script to be added to the footer
+
+    <script>
+        $(document).ready(function () {
+            $("ul#navigation").addClass('mobile-hidden');
+            $(".js-toggle-nav .js-header-toggle").click(function () {
+                event.preventDefault();
+                $("ul#navigation").toggleClass('mobile-hidden');
+                $("#global-nav").toggleClass('open');
+            });
+        });
+    </script>
+#}
+{% macro addPredefinedNavbar(userDisplayText, chsMonitorGuiUrl, lang) %}
+    {% set signInHref = "/signin" %}
+    {% set menuItems = [] %}
+    {% set menuItems = (menuItems.push({
+        id: "your-companies",
+        href: "/your-companies",
+        displayText: lang.yourCompanies if lang else "Your companies"
+    }), menuItems) %}
+    {% set menuItems = (menuItems.push({
+        id: "authorised-agent",
+        href: "/authorised-agent",
+        displayText: lang.authorisedAgent if lang else "Authorised agent"
+    }), menuItems) %}
+    {% set menuItems = (menuItems.push({
+        id: "your-filings",
+        href: "/user/transactions",
+        displayText: lang.yourFilings if lang else "Your filings"
+    }), menuItems) %}
+    {% set menuItems = (menuItems.push({
+        id: "companies-you-follow",
+        href: chsMonitorGuiUrl,
+        displayText: lang.companiesYouFollow if lang else "Companies you follow"
+    }), menuItems) %}
+    {% set menuItems = (menuItems.push({
+        id: "basket",
+        href: "/basket",
+        displayText: lang.basket if lang else 
+            "Basket"
+    }), menuItems) %}
+    {% set menuItems = (menuItems.push({
+        id: "your-details",
+        href: "/user/account",
+        displayText: lang.manageAccount if lang else "Manage account"
+    }), menuItems) %}
+    {% set menuItems = (menuItems.push({
+        id: "user-signout",
+        href: "/signout",
+        displayText: lang.signOut if lang else "Sign out"
+    }), menuItems) %}
```

Comment:
Instead of all those push, why not defining directly the array as it is:
```
{% set menuItems = [
        { id: "your-companies", href: "/your-companies", displayText: lang.yourCompanies or "Your companies" },
         ......
        { id: "user-signout", href: "/signout", displayText: lang.signOut or "Sign out" }
    ] %}
```